src/main/webapp/app/exam/participate/exam-participation.component.ts

```diff
@@ -191,6 +192,8 @@ export class ExamParticipationComponent implements OnInit, OnDestroy, ComponentC
                         participation.submissions.forEach((submission) => {
                             submission.isSynced = true;
                         });
+                    } else if (exercise.type === ExerciseType.PROGRAMMING) {
+                        participation.submissions.push(new ProgrammingSubmission());
```

Comment:
why are you doing this? Can you add some comments?

Reply:
We use the isSynced flag of the Submission class to show to appropriate status in the navigation bar header of the student exam conduction. Without this, the navigation bar header is not uploaded. For programming exercises however, when the student first loads the student exam, programming exercises do not have a submission in their participation, even though their status is correctly set to 'initialised'. 
This led to programming exercises which do not have a student submission yet, would not update the navigation bar header. 
An easy way around this is to set an initial programming submission. This would be overridden as soon as the student commits something. After the user commits for the first time for a given exercise, the real submission is loaded.